test/streams.ts

```diff
@@ -0,0 +1,254 @@
+      done();
+    });
+  });
+
+  it('should create read stream and read synchronously using Transform stream', done => {
+    const keyFrom = 0;
+    const keyTo = 1000;
+
+    service.setService({
+      ReadRows: readRowsImpl(keyFrom, keyTo),
+    });
+
+    let receivedRowCount = 0;
+    let lastKeyReceived: number | undefined;
+
+    // BigTable stream
+    const readStream = table.createReadStream();
+
+    // Transform stream
+    const transform = new Transform({
+      objectMode: true,
+      transform: (row, _encoding, callback) => {
+        callback(null, row);
+      },
+    });
+
+    // Final stream
+    const passThrough = new PassThrough({
+      objectMode: true,
+    });
+
+    passThrough.on('error', (err: GoogleError) => {
+      done(err);
+    });
+    passThrough.on('data', (row: Row) => {
+      ++receivedRowCount;
+      const key = parseInt(row.id);
+      if (lastKeyReceived && key <= lastKeyReceived) {
+        done(new Error('Test error: keys are not in order'));
+      }
+      lastKeyReceived = key;
+      debugLog(`received row key ${key}`);
+    });
+    passThrough.on('end', () => {
+      assert.strictEqual(receivedRowCount, keyTo - keyFrom);
+      assert.strictEqual(lastKeyReceived, keyTo - 1);
+      done();
+    });
+
+    pipeline(readStream, transform, passThrough, () => {});
+  });
+
+  // TODO(@alexander-fenster): enable after https://github.com/googleapis/nodejs-bigtable/issues/607 is fixed
+  it.skip('should create read stream and read asynchronously using Transform stream', done => {
+    const keyFrom = 0;
+    const keyTo = 1000;
+
+    service.setService({
+      ReadRows: readRowsImpl(keyFrom, keyTo),
+    });
+
+    let receivedRowCount = 0;
+    let lastKeyReceived: number | undefined;
+
+    // BigTable stream
+    const readStream = table.createReadStream();
+
+    // Transform stream
+    const transform = new Transform({
+      objectMode: true,
+      transform: (row, _encoding, callback) => {
+        setTimeout(() => {
+          callback(null, row);
+        }, 0);
+      },
+    });
+
+    // Final stream
+    const passThrough = new PassThrough({
+      objectMode: true,
+    });
+
+    passThrough.on('error', (err: GoogleError) => {
+      done(err);
+    });
+    passThrough.on('data', (row: Row) => {
+      ++receivedRowCount;
+      const key = parseInt(row.id);
+      if (lastKeyReceived && key <= lastKeyReceived) {
+        done(new Error('Test error: keys are not in order'));
+      }
+      lastKeyReceived = key;
+      debugLog(`received row key ${key}`);
+    });
+    passThrough.on('end', () => {
+      assert.strictEqual(receivedRowCount, keyTo - keyFrom);
+      assert.strictEqual(lastKeyReceived, keyTo - 1);
+      done();
+    });
+
+    pipeline(readStream, transform, passThrough, () => {});
+  });
+
+  it('should be able to stop reading from the read stream', done => {
+    const keyFrom = 0;
+    const keyTo = 1000;
+    const stopAfter = 42;
```

Comment:
Consider adding a comment about why 42 is the chosen "stopAfter" value if there's anything of note about it

Reply:
It's random (and also the answer to life the universe and everything). I'll note that!